Exceptions raised in shell will not be caught when not in TTY

vmikki opened the pull request:
This PR changes the behaivour of Nameko shell so it doesn't catch exceptions when used in non-TTY mode. A use case when this would be useful is when nameko shell used for cron jobs and you need to know wether the executed command was successful or not. For example after executing the following command:
```sh
$ echo "n.rpc.other_service.foo()" | nameko shell
```
the exit code will be set to 1 if the execution of the remote method failed.

mattbennett commented:
We pipe input to the shell quite frequently, as a way of essentially writing small scripts that use the `n` helper. It is helpful to bubble exceptions up to the process level in this case.

The alternative is to write a standalone script, but you have to reimplement all the config parsing and helper setup.

vmikki commented on nameko/cli/code.py:
```diff
@@ -0,0 +1,36 @@
+from code import InteractiveConsole
```
I have added `from __future__ import absolute_import` and now it works in Python 2.7.

vmikki commented:
I have fixed the broken builds and now my PR is ready to be reviewed.
I had to use subprocesses in order to reliably test the new feature and the code covered by those tests is marked as not covered in coverage. The possible solution for that would have been a bit hacky and unreliable so I just added `no cover` instead, I hope it's fine.

mattbennett commented on nameko/cli/code.py:
```diff
@@ -0,0 +1,38 @@
+from code import InteractiveConsole
+
+
+class RaisingInteractiveConsole(InteractiveConsole):  # pragma: no cover
```
You might be able to `no cover` the whole module in one go.

I think it'd also be worth putting a little note explaining why it's excluded from coverage.

vmikki replied:
Is there a way to add `no cover` to the whole module? I think excluding the whole file can only be done by specifying the `omit` parameter for coverage but in that case it will not be visible in the module that it is omitted from coverage. I agree that it should be explained why `no cover` is there but perhaps it would be better to be explicit and keep the current `no cover`s with some comment next to them.
